scripts/ngi_aggregate_statistics

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+""" 
+This is a script to create NGI aggregate report.
```

Comment:
That's basically what I mean - except instead of a utility script, have it as a utility function within the base class (which everything should be using).

Nice idea to run it before plotting stuff. The only problem I can see with doing this is that currently the plotting is tied to getting the information that the plots need. So if we ever want any of this information in the `.csv`, it won't be generated yet. But I agree that it could be nice to let that run and then `Ctrl-C` once the plots have started.

Maybe we can move the plot calls out into their own function and call this a little later? That could be tidy.